permissions/src/iosMain/kotlin/location/LocationPermissionManager.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ Copyright (c) 2020. Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.permissions.location
+
+import com.splendo.kaluga.permissions.IOSPermissionsHelper
+import com.splendo.kaluga.permissions.Permission
+import com.splendo.kaluga.permissions.PermissionManager
+import com.splendo.kaluga.permissions.PermissionState
+import platform.CoreLocation.CLAuthorizationStatus
+import platform.CoreLocation.CLLocationManager
+import platform.CoreLocation.CLLocationManagerDelegateProtocol
+import platform.Foundation.NSBundle
+import platform.darwin.NSObject
+
+actual class LocationPermissionManager(
+    private val bundle: NSBundle,
+    actual val location: Permission.Location,
+    stateRepo: LocationPermissionStateRepo
+) : PermissionManager<Permission.Location>(stateRepo) {
+
+    private val locationManager = CLLocationManager()
+    private val authorizationStatus = {
+        CLLocationManager.authorizationStatus().toAuthorizationStatus(location.background)
+    }
+
+    private val delegate = object : NSObject(), CLLocationManagerDelegateProtocol {
+
+        override fun locationManager(manager: CLLocationManager, didChangeAuthorizationStatus: CLAuthorizationStatus /* = kotlin.Int */) {
+            val locationPermissionManager = this@LocationPermissionManager
+            IOSPermissionsHelper.handleAuthorizationStatus(didChangeAuthorizationStatus.toAuthorizationStatus(locationPermissionManager.location.background), locationPermissionManager)
+        }
+
+    }
+
+    override suspend fun requestPermission() {
+        val locationDeclarations = mutableListOf("NSLocationWhenInUseUsageDescription")
+        if (location.background) {
+            locationDeclarations.addAll(listOf("NSLocationAlwaysAndWhenInUseUsageDescription", "NSLocationAlwaysUsageDescription"))
+        }
+        if (IOSPermissionsHelper.checkDeclarationInPList(bundle, *locationDeclarations.toTypedArray()).isEmpty()) {
```

Comment:
And here is confusion, you check declarations and have a list of this declarations but do main flow when it's empty